modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/files_query.rb

```diff
@@ -58,12 +58,17 @@ def query(parent)
 
     private
 
+    def escape_whitespace(value)
```

Comment:
What do you think of using this sanitizer instead?

```ruby
ActiveStorage::Filename.new("foo/bar.jpg").sanitized
```

https://api.rubyonrails.org/classes/ActiveStorage/Filename.html#method-i-sanitized

Probably there are a few more edge cases than just a whitespace. Also can we test this?

Reply:
this class is unit tested. I could add a case for the user id containing whitespaces.

Not sure though, about the sanitizer. It actually switches characters, that could be an issue. I'll have a look into it. In general we do not change the values of user id and parent folder, just make them url safe.